GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -497,41 +494,30 @@ void UpdateGraph(int fromIndex, in int toIndex)
 
                 while (rowIndex < toIndex)
                 {
-                    lock (_graphModel)
+                    // Cache the next item
+                    if (!_graphModel.CacheTo(rowIndex))
                     {
-                        // Cache the next item
-                        if (!_graphModel.CacheTo(rowIndex))
+                        Debug.WriteLine("Cached item FAILED {0}", rowIndex);
+                        lock (_backgroundThread)
```

Comment:
What is the purpose of locking on the `_backgroundThread`? It seems to be suspicious to lock on a reference that may be modified.

Reply:
These locks can be removed. Done.